### ISSUE TITLE
Flaky TestIngester_Starting

### ISSUE THREAD

**Issue by narqo** [grafana/mimir#7806]
The [build failed][1], due to flaky test:

```
--- FAIL: TestIngester_Starting (12.01s)
    --- FAIL: TestIngester_Starting/if_starting()_of_an_ingester_with_ring_state_LEAVING_runs_into_an_error,_its_lifecycler_terminates,_and_its_ring_state_is_LEAVING (3.00s)
        ingester_test.go:10664: 
            	Error Trace:	/__w/mimir/mimir/pkg/ingester/ingester_test.go:10664
            	            				/__w/mimir/mimir/pkg/ingester/ingester_test.go:10620
            	Error:      	Target error should be in err chain:
            	            	expected: "this is an error"
            	            	in chain: "failed to start ingester subservices before partition reader: not healthy, 3 terminated, 0 failed: []"
            	            		"failed to start ingester subservices before partition reader: not healthy, 3 terminated, 0 failed: []"
            	            		"not healthy, 3 terminated, 0 failed: []"
            	Test:       	TestIngester_Starting/if_starting()_of_an_ingester_with_ring_state_LEAVING_runs_into_an_error,_its_lifecycler_terminates,_and_its_ring_state_is_LEAVING
level=info msg="uploading new block to long-term storage" block=00000000010000000000000000
level=debug msg="uploaded file" from=/tmp/TestShipper_DeceivingUploadErrors2722881139/001/00000000010000000000000000/index dst=00000000010000000000000000/index bucket="fs: /tmp/TestShipper_DeceivingUploadErrors2722881139/002"
level=error msg="uploading new block to long-term storage failed" block=00000000010000000000000000 err="upload meta file: base name matches, will fail upload"
```

[1]: https://github.com/grafana/mimir/actions/runs/8554926103/job/23441283893

**Comment by charleskorn** [comment]
[Another example](https://github.com/grafana/mimir/actions/runs/8673169743/job/23784240718?pr=7849#step:8:67)

```
--- FAIL: TestIngester_Starting (12.01s)
    --- FAIL: TestIngester_Starting/if_starting()_of_an_ingester_with_ring_state_LEAVING_runs_into_an_error,_its_lifecycler_terminates,_and_its_ring_state_is_LEAVING (3.00s)
        ingester_test.go:10664: 
            	Error Trace:	/__w/mimir/mimir/pkg/ingester/ingester_test.go:10664
            	            				/__w/mimir/mimir/pkg/ingester/ingester_test.go:10620
            	Error:      	Target error should be in err chain:
            	            	expected: "this is an error"
            	            	in chain: "failed to start ingester subservices before partition reader: not healthy, 0 terminated, 0 failed: []"
            	            		"failed to start ingester subservices before partition reader: not healthy, 0 terminated, 0 failed: []"
            	            		"not healthy, 0 terminated, 0 failed: []"
            	Test:       	TestIngester_Starting/if_starting()_of_an_ingester_with_ring_state_LEAVING_runs_into_an_error,_its_lifecycler_terminates,_and_its_ring_state_is_LEAVING
```

**Comment by charleskorn** [comment]
@duricanikolic looks like you added this test in https://github.com/grafana/mimir/pull/6923, could you please take a look at this?

**Comment by duricanikolic** [comment]
Hi @charleskorn 
I will have a look during this week.

**Comment by duricanikolic** [comment]
From the first look that I had, I saw a message that was introduced recently (#7769):
```
failed to start ingester subservices before partition reader
```

**Comment by 56quarters** [comment]
Hopefully fixed by #7917 :crossed_fingers:
